v2.1/build-a-ruby-app-with-cockroachdb.md

```diff
@@ -65,15 +65,13 @@ Initial balances:
 
 Next, use the following code to again connect as the `maxroach` user but this time execute a batch of statements as an atomic transaction to transfer funds from one account to another, where all included statements are either committed or aborted.
 
-Download the <a href="https://raw.githubusercontent.com/cockroachdb/docs/master/_includes/app/txn-sample.rb" download><code>txn-sample.rb</code></a> file, or create the file yourself and copy the code into it.
+Download the <a href="https://raw.githubusercontent.com/cockroachdb/docs/master/_includes/v2.1/app/txn-sample.rb" download><code>txn-sample.rb</code></a> file, or create the file yourself and copy the code into it.
 
-{{site.data.alerts.callout_info}}
-With the default `SERIALIZABLE` isolation level, CockroachDB may require the [client to retry a transaction](transactions.html#transaction-retries) in case of read/write contention. CockroachDB provides a generic **retry function** that runs inside a transaction and retries it as needed. You can copy and paste the retry function from here into your code.
-{{site.data.alerts.end}}
+{% include v2.1/client-transaction-retry.md %}
 
 {% include copy-clipboard.html %}
 ~~~ ruby
-{% include app/txn-sample.rb %}
+{% include v2.1/app/txn-sample.rb %}
 ~~~
 
 Then run the code:
```

Comment:
Getting this error:

/usr/local/lib/ruby/gems/2.4.0/gems/pg-1.0.0/lib/pg.rb:56:in `initialize': fe_sendauth: no password supplied (PG::ConnectionBad)
	from /usr/local/lib/ruby/gems/2.4.0/gems/pg-1.0.0/lib/pg.rb:56:in `new'
	from /usr/local/lib/ruby/gems/2.4.0/gems/pg-1.0.0/lib/pg.rb:56:in `connect'
	from txn-sample.rb:36:in `<main>'